ktor-client/ktor-client-cio/jvm/src/io/ktor/client/engine/cio/Endpoint.kt

```diff
@@ -177,7 +193,27 @@ internal class Endpoint(
         }
 
         connections.decrementAndGet()
-        throw FailToConnectException()
+
+        throw when (timeoutFails) {
```

Comment:
Consider extracting reason here. It's much easier to read a single line `throw`

Reply:
Done.